arelle/FileSource.py

```diff
@@ -200,7 +200,6 @@ def open(self, reloadCache: bool = False) -> None:
                 return  # an error should have been logged
             if self.isZip:
                 try:
-                    assert self.cntlr is not None
```

Comment:
arbitrary line for discussion:

> What code path does this happen in? It seems odd?

From building a [xule rule set](https://github.com/xbrlus/xule/blob/60354a037977f0e0442f0c92b265fc594501b99c/plugin/xule/XuleRuleSet.py#L159).

Reply:
Huh, is that the expected usage there? I haven't ever seen someone loading using the filesource directly

Reply:
Is this path hit when loading the XULE validation zips?

Reply:
> Huh, is that the expected usage there? I haven't ever seen someone loading using the filesource directly

Yes, the file stream methods specifically check for and handle the none cntlr case.